src/plugins/webpack/index.ts

```diff
@@ -70,15 +70,26 @@ const findWebpackDependencies: GenericPluginCallback = async (configFilePath, {
       const dependencies = (config.module?.rules?.flatMap(resolveRuleSetDependencies) ?? []).map(loader =>
         loader.replace(/\?.*/, '')
       );
-      const entries = (
-        cfg.entry
-          ? typeof cfg.entry === 'string'
-            ? [cfg.entry]
-            : Array.isArray(cfg.entry)
-            ? cfg.entry
-            : Object.values(cfg.entry).map(entry => (typeof entry === 'string' ? entry : entry.filename))
-          : []
-      ).map(entry => (config.context ? join(config.context, entry) : entry));
+      let entries: string[] = [];
+
+      if (typeof cfg.entry === "string") entries.push(cfg.entry);
+      else if (Array.isArray(cfg.entry)) entries.push(...cfg.entry);
+      else if (typeof cfg.entry === "object") {
+        Object.values(cfg.entry).map(entry => {
+          if (typeof entry === "string") entries.push(entry);
+          else if (Array.isArray(entry)) entries.push(...entry);
+          else if (typeof entry === "function") entries.push((entry as () => string)());
+          else if (entry && typeof entry === "object" && "filename" in entry) entries.push(entry["filename"] as string);
+          else {
+            // TODO: warn about unrecognized single entry in object
```

Comment:
It's not Knip's responsibility to validate/lint the configuration of external tooling. The plugins just try to find dependencies.

Reply:
I wouldn't say it's about validation or linting, but instead making it easier to detect when Webpack changes/extends its configuration schema and the plugin doesn't recognize it. Rather than crash as happened in my case :P

I left this note in case `knip` has a preferred way of surfacing such warnings. But I can remove the comments if you like.

Reply:
There's no handling of such config issues (I'd leave that to e.g. Webpack in this case), other than handling valid configuration.

Reply:
Right -- I'm not sure you've understood me. The scenario here is that some configuration is perfectly valid according to Webpack, but hasn't been handled in Knip for some reason. So a warning would be useful to make it clear that Knip needs to handle a new case.

But it sounds like you don't want it, so I'll remove these branches.

Reply:
Thanks, what you describe is the ongoing process of what is Knip is all about, to keep iterating and improving based on fixes like yours. Unless you already have a specific case in mind that you know should be handled, the comment does not add much for me.

Reply:
Oh well the suggestion was to turn the comment into a `console.warn`, or some other suitable way of surfacing a warning.

That way the next person who runs into the same scenario won't have to decipher an opaque error. (My error was an `undefined` string deep inside some stacktrace.)

Reply:
But no worries either way, my use-case works after this fix so I'm happy.

Reply:
Exception handling in general is something that should be improved indeed!